File: DigiMox/app/src/main/java/com/digimox/fragments/DMSelectedMenuFragment.java
package com.digimox.fragments;

import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.digimox.R;
import com.digimox.activity.DMHomeActivity;
import com.digimox.adapters.DMSelectedItemAdapter;
import com.digimox.app.DMAppConstants;
import com.digimox.models.response.DMSubCategory;
import com.digimox.utils.DMDataBaseHelper;
import com.digimox.utils.DMUtils;

import java.util.ArrayList;
import java.util.Collections;

/**
 * Created by Deepesh on 31-Dec-15.
 */
public class DMSelectedMenuFragment extends DMBaseFragment implements View.OnClickListener {
    private View selectedView;
    private RecyclerView subCategoryList;
    private ArrayList<DMSubCategory> dmSubCategories;
    private DMDataBaseHelper dmDataBaseHelper;
    private DMSelectedItemAdapter dmSelectedItemAdapter;

    public DMSelectedMenuFragment() {
    }

    public void setDmSubCategories(ArrayList<DMSubCategory> dmSubCategories) {
        this.dmSubCategories = dmSubCategories;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        selectedView = inflater.inflate(R.layout.activity_selected_list, container, false);
        initViews();
        setOnClickListener();
//        ((DMHomeActivity) getActivity()).showBackButton();
        if (null != dmSubCategories && dmSubCategories.size() > 0) {
            initList(dmSubCategories);
        }
        return selectedView;
    }


    private void initViews() {
        subCategoryList = (RecyclerView) selectedView.findViewById(R.id.sub_category_list);
        subCategoryList.setHasFixedSize(true);
        checkOrientation();
//        subCategoryList.addItemDecoration(
//                new DMDividerItemDecoration(getActivity(), R.drawable.sublist_divider));
        setMenuCount();
    }

    private void checkOrientation() {
        int orientation = DMUtils.getScreenOrientation(getActivity());
        if (DMAppConstants.PORTRAIT == orientation) {
            subCategoryList.setLayoutManager(new GridLayoutManager(getActivity(), 1));
        } else {
            subCategoryList.setLayoutManager(new GridLayoutManager(getActivity(), 2));
        }
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        int orientation = newConfig.orientation;
        switch (orientation) {
            case Configuration.ORIENTATION_LANDSCAPE:
                subCategoryList.setLayoutManager(new GridLayoutManager(getActivity(), 2));
                break;
            case Configuration.ORIENTATION_PORTRAIT:
                subCategoryList.setLayoutManager(new GridLayoutManager(getActivity(), 1));
                break;
        }
    }

    private void initList(ArrayList<DMSubCategory> dmSubCategories) {
        dmSubCategories.removeAll(Collections.singleton(null));
        dmSubCategories.add(null);
        dmSelectedItemAdapter = new DMSelectedItemAdapter(getActivity(), dmSubCategories, this);
        subCategoryList.setAdapter(dmSelectedItemAdapter);
    }

    public void clearData() {
        dmDataBaseHelper = new DMDataBaseHelper(getActivity());
        dmDataBaseHelper.openDataBase();
        dmDataBaseHelper.deleteTable();
        dmDataBaseHelper.close();
        ((DMHomeActivity) getActivity()).setMeuCount("0");
    }

    public void setMenuCount() {
        dmDataBaseHelper = new DMDataBaseHelper(getActivity());
        dmDataBaseHelper.openDataBase();
        dmSubCategories = dmDataBaseHelper.getAddedList();
        int count = dmSubCategories.size();
        ((DMHomeActivity) getActivity()).setMeuCount(String.valueOf(count));
        dmDataBaseHelper.close();
        if (count == 0) {
            selectedView.findViewById(R.id.no_item).setVisibility(View.VISIBLE);
        }
    }

    private void setOnClickListener() {
        selectedView.findViewById(R.id.clear_list).setOnClickListener(this);

    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.clear_list:
                if (null != dmSelectedItemAdapter) {
                    clearData();
                    dmSelectedItemAdapter.remove();
                    selectedView.findViewById(R.id.no_item).setVisibility(View.VISIBLE);
                }
                break;

        }

    }

    public void showItemDetails(DMSubCategory dmSubCategory) {
        if (null != getActivity()) {
            final android.support.v4.app.FragmentManager fm = getActivity().getSupportFragmentManager();
            if (null != fm) {
                DMSelectedItemDetailFragment dmSelectedItemDetailFragment = new DMSelectedItemDetailFragment();
                dmSelectedItemDetailFragment.setDmSubCategory(dmSubCategory);
                final android.support.v4.app.FragmentTransaction ft = fm.beginTransaction();
                ft.setCustomAnimations(R.anim.slide_in_right, R.anim.slide_out_left, R.anim.slide_in_right, R.anim.slide_out_left)
                        .add(R.id.fragment_container, dmSelectedItemDetailFragment, "selectedItemDetailsFragment")
                        .addToBackStack("selectedItemDetailsFragment")
                        .commit();
            }
        }
    }
}

File: DigiMox/app/src/main/java/com/digimox/fragments/DMSelectedMenuFragment.kt
package com.digimox.fragments

import android.content.res.Configuration
import android.os.Bundle
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.digimox.R
import com.digimox.activity.DMHomeActivity
import com.digimox.adapters.DMSelectedItemAdapter
import com.digimox.app.DMAppConstants
import com.digimox.models.response.DMSubCategory
import com.digimox.utils.DMDataBaseHelper
import com.digimox.utils.DMUtils

import java.util.ArrayList
import java.util.Collections

/**
 * Created by Deepesh on 31-Dec-15.
 */
class DMSelectedMenuFragment : DMBaseFragment(), View.OnClickListener {
    private var selectedView: View? = null
    private var subCategoryList: RecyclerView? = null
    private var dmSubCategories: ArrayList<DMSubCategory?>? = null
    private var dmDataBaseHelper: DMDataBaseHelper? = null
    private var dmSelectedItemAdapter: DMSelectedItemAdapter? = null

    fun setDmSubCategories(dmSubCategories: ArrayList<DMSubCategory?>?) {
        this.dmSubCategories = dmSubCategories
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        selectedView = inflater.inflate(R.layout.activity_selected_list, container, false)
        initViews()
        setOnClickListener()
        //        ((DMHomeActivity) getActivity()).showBackButton();
        if (null != dmSubCategories && dmSubCategories!!.size > 0) {
            initList(dmSubCategories!!)
        }
        return selectedView
    }


    private fun initViews() {
        subCategoryList = selectedView!!.findViewById<View>(R.id.sub_category_list) as RecyclerView
        subCategoryList!!.setHasFixedSize(true)
        checkOrientation()
        //        subCategoryList.addItemDecoration(
        //                new DMDividerItemDecoration(getActivity(), R.drawable.sublist_divider));
        setMenuCount()
    }

    private fun checkOrientation() {
        val orientation = DMUtils.getScreenOrientation(activity)
        if (DMAppConstants.PORTRAIT == orientation) {
            subCategoryList!!.layoutManager = GridLayoutManager(activity, 1)
        } else {
            subCategoryList!!.layoutManager = GridLayoutManager(activity, 2)
        }
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        val orientation = newConfig.orientation
        when (orientation) {
            Configuration.ORIENTATION_LANDSCAPE -> subCategoryList!!.layoutManager = GridLayoutManager(activity, 2)
            Configuration.ORIENTATION_PORTRAIT -> subCategoryList!!.layoutManager = GridLayoutManager(activity, 1)
        }
    }

    private fun initList(dmSubCategories: ArrayList<DMSubCategory?>) {
        dmSubCategories.filterNotNull()
        dmSubCategories.add(null)
        dmSelectedItemAdapter = DMSelectedItemAdapter(activity!!, dmSubCategories, this)
        subCategoryList!!.adapter = dmSelectedItemAdapter
    }

    fun clearData() {
        dmDataBaseHelper = DMDataBaseHelper(activity)
        dmDataBaseHelper!!.openDataBase()
        dmDataBaseHelper!!.deleteTable()
        dmDataBaseHelper!!.close()
        (activity as DMHomeActivity).setMeuCount("0")
    }

    fun setMenuCount() {
        dmDataBaseHelper = DMDataBaseHelper(activity)
        dmDataBaseHelper!!.openDataBase()
        dmSubCategories = dmDataBaseHelper!!.addedList
        val count = dmSubCategories!!.size
        (activity as DMHomeActivity).setMeuCount(count.toString())
        dmDataBaseHelper!!.close()
        if (count == 0) {
            selectedView!!.findViewById<View>(R.id.no_item).visibility = View.VISIBLE
        }
    }

    private fun setOnClickListener() {
        selectedView!!.findViewById<View>(R.id.clear_list).setOnClickListener(this)

    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.clear_list -> if (null != dmSelectedItemAdapter) {
                clearData()
                dmSelectedItemAdapter!!.remove()
                selectedView!!.findViewById<View>(R.id.no_item).visibility = View.VISIBLE
            }
        }

    }

    fun showItemDetails(dmSubCategory: DMSubCategory?) {
        if (null != activity) {
            val fm = activity!!.supportFragmentManager
            if (null != fm) {
                val dmSelectedItemDetailFragment = DMSelectedItemDetailFragment()
                dmSelectedItemDetailFragment.setDmSubCategory(dmSubCategory)
                val ft = fm.beginTransaction()
                ft.setCustomAnimations(R.anim.slide_in_right, R.anim.slide_out_left, R.anim.slide_in_right, R.anim.slide_out_left)
                        .add(R.id.fragment_container, dmSelectedItemDetailFragment, "selectedItemDetailsFragment")
                        .addToBackStack("selectedItemDetailsFragment")
                        .commit()
            }
        }
    }
}

